src/components/Year/Year.tsx

```diff
@@ -2,466 +2,438 @@ import { useEffect, useState } from "react";
+        {
+            dataField: "Actions",
+            text: "Actions",
+            formatter: (cell: any, row: any, rowIndex: any, formatExtraData: any) => {
+                return (
+                    <Button
+                        variant="danger"
+                        onClick={() => {
+                            setdatatoDelete({
+                                index: row.index - 1,
+                                year: row.academic_year,
+                                id: row.year_id,
+                            });
+                            handleShow();
+                        }}
+                    >
+                        Delete
+                    </Button>
+                );
+            },
+            sort: true,
+        },
+    ];
 
-  const getAllAcademicYear = () => {
-    getAccessToken();
-    axios
-      .get(`${baseUrl}year`)
-      .then((res: any) => {
-        res.data.data.map((data: any, index: any) => {
-          data.index = index + 1;
-        });
-        setAllAcademicYear(res.data.data);
-        setSpinnerLoad(false);
-      })
-      .catch((e: any) => {
-        console.log(e);
-        setSpinnerLoad(false);
-      });
-  };
+    // const getAllAcademicYear = () => {
+    //     getAccessToken();
+    //     axios
+    //         .get(`${baseUrl}year`)
+    //         .then((res: any) => {
+    //             res.data.data.map((data: any, index: any) => {
+    //                 data.index = index + 1;
+    //             });
+    //             setAllAcademicYear(res.data.data);
+    //             setSpinnerLoad(false);
+    //         })
+    //         .catch((e: any) => {
+    //             console.log(e);
+    //             setSpinnerLoad(false);
+    //         });
+    // };
 
-  //Modal Popup
-  const [show, setShow] = useState(false);
-  const handleClose = () => {
-    setShow(false);
-    deleteAnAcademicYear(datatoDelete.id, datatoDelete.index);
-  };
-  const SuddenhandleClose = () => {
-    setShow(false);
-    setdatatoDelete({});
-  };
-  const handleShow = () => {
-    setShow(true);
-  };
-  const datatoFilterNull: any =
-    allAcademicYear &&
-    allAcademicYear.length &&
-    allAcademicYear.sort().map((data: any) => {
-      let keys = Object.keys(data);
-      keys.map((key: any) => {
-        data[key] = data[key] == null ? "" : data[key];
-      });
-      return data;
-    });
-  const dataSearch: any =
-    datatoFilterNull &&
-    datatoFilterNull.length &&
-    datatoFilterNull.sort().filter((data: any) => {
-      return Object.keys(data).some((key) =>
-        data[key]
-          .toString()
-          .toLowerCase()
-          .includes(filter.toString().toLowerCase())
-      );
-    });
+    //Modal Popup
+    const [show, setShow] = useState(false);
+    const handleClose = () => {
+        setShow(false);
+        dispatch(deleteYear(datatoDelete.id, datatoDelete.index));
```

Comment:
please check the line

Reply:
I checked the line here it is used to dispatch the delete action for a specific id and the index of the id.